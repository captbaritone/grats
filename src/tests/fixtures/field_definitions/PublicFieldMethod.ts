/** @gqlType */
export class User {
  /** @gqlField */
  public greet(): string {
    return "Hello";
  }
}
