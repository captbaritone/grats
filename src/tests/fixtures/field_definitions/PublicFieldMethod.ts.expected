-----------------
INPUT
----------------- 
/** @gqlType */
export class User {
  /** @gqlField */
  public greet(): string {
    return "Hello";
  }
}

-----------------
OUTPUT
-----------------
-- SDL --
type User {
  greet: String @metadata
}
-- TypeScript --
import { GraphQLSchema, GraphQLObjectType, GraphQLString } from "graphql";
export function getSchema(): GraphQLSchema {
    const UserType: GraphQLObjectType = new GraphQLObjectType({
        name: "User",
        fields() {
            return {
                greet: {
                    name: "greet",
                    type: GraphQLString
                }
            };
        }
    });
    return new GraphQLSchema({
        types: [UserType]
    });
}
